core/dbt/contracts/util.py

```diff
@@ -342,9 +226,12 @@ def read_and_check_versions(cls, path: str):
                         expected=str(cls.dbt_schema_version),
                         found=previous_schema_version,
                     )
-        if get_manifest_schema_version(data) <= 8:
-            data = upgrade_manifest_json(data)
-        return cls.from_dict(data)  # type: ignore
+
+        return cls.upgrade_schema_version(data)
+
+    @classmethod
+    def upgrade_schema_version(cls, data):
+        return cls.from_dict(data)
```

Comment:
Somewhat of a repeat of my other comment. Does `from_dict` actually do an upgrade?

Reply:
No. This is just the "base" implementation of "upgrade_schema_version" which is to do nothing to the schema version, but does return an object from the dictionary that was loaded from the json.  It's only the manifest version of this method which actually does an "upgrade".

Reply:
Ah gotcha! So this method exists because a default implementation of "upgrade_schema_version" must exist. That makes a ton of sense, and was the bit of context I was missing. Thank you 🙏